File: aula-01/js/Ex07.js
/**
 * Verifica o produto de dois inteiros a e b utilizando soma
 * @param {Number} a variável multiplicando
 * @param {Number} b variável multiplicador
 * @throws {RangeError} caso a e b sejam menores que 0
 * @returns {Number} retorna o produto de dois inteiros
 */
module.exports = produto;
function produto(a, b) {

    if (a < 0) {
        throw new RangeError("Intervalo de a inválido");
    }

    if (b < 0) {
        throw new RangeError("Intervalo de b inválido");
    }

    let totalParcelas = a;
    let parcela = b;
    if (b < a) {
        totalParcelas = b;
        parcela = a;
    }

    let i = 1;
    let s = 0;
    while (i <= totalParcelas) {
        s = s + parcela;
        i = i + 1;
    }

    return s;
}

File: aula-01/java/Ex07.java
public class Ex07 {

    public int Produto(int a, int b) {
        
        if (a < 0) {
            throw new IllegalArgumentException("Intervalo de a inválido");
        }

        if (b < 0) {
            throw new IllegalArgumentException("Intervalo de a inválido");
        }

        int totalParcelas = a;
        int parcela = b;

        if (a > b){
            totalParcelas = b;
            parcela = a;
        }

        int s = 0;
        int i = 1;

        while (i <= totalParcelas) {
            s = s + parcela;
        }
        
        return s;
    }
}